abci/example/kvstore/kvstore_test.go

```diff
@@ -536,17 +545,17 @@ func newKvApp(ctx context.Context, t *testing.T, genesisHeight int64, opts ...Op
 	return app
 }
 
-func assertRespInfo(t *testing.T, expectHeight int64, expectAppHash tmbytes.HexBytes, actual types.ResponseInfo, msgs ...interface{}) {
+func assertRespInfo(t *testing.T, expectLastBlockHeight int64, expectAppHash tmbytes.HexBytes, actual types.ResponseInfo, msgs ...interface{}) {
 	t.Helper()
 
 	if expectAppHash == nil {
 		expectAppHash = make(tmbytes.HexBytes, tmcrypto.DefaultAppHashSize)
 	}
 	expected := types.ResponseInfo{
-		LastBlockHeight:  expectHeight,
+		LastBlockHeight:  expectLastBlockHeight,
 		LastBlockAppHash: expectAppHash,
 		Version:          version.ABCIVersion,
-		AppVersion:       ProtocolVersion,
+		AppVersion:       uint64(expectLastBlockHeight + 1),
```

Comment:
In our case, it will be max available version.

Reply:
This is just a test. I modified kvstore to put current height into app version, for testing purposes.

Do we still need proposed_app_version mechanism? Maybe we should also deprecate it?

Reply:
Yes, we still need it